src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,346 @@
+                )
+        );
+
+        mediaTypes.add(
+                jsonObjectBuilder().add(
+                        "href",
+                        systemConfig.getDataverseSiteUrl() + "/api/datasets/export?exporter=schema.org&persistentId=" + ds.getProtocol() + ":" + ds.getAuthority() + "/" + ds.getIdentifier()
+                ).add(
+                        "type",
+                        "application/json+ld"
+                )
+        );
+        JsonArrayBuilder linksetJsonObj = Json.createArrayBuilder();
+
+        JsonObjectBuilder mandatory;
+        if (useDefaultFileType) {
+            mandatory = jsonObjectBuilder()
+                    .add("anchor", landingPage)
+                    .add("cite-as", Json.createArrayBuilder().add(jsonObjectBuilder().add("href", ds.getPersistentURL())))
+                    .add("type", Json.createArrayBuilder()
+                            .add(jsonObjectBuilder().add("href", "https://schema.org/AboutPage"))
+                            .add(jsonObjectBuilder().add("href", defaultFileTypeValue))
+                    );
+        } else {
+            mandatory = jsonObjectBuilder()
+                    .add("anchor", landingPage)
+                    .add("cite-as", Json.createArrayBuilder().add(jsonObjectBuilder().add("href", ds.getPersistentURL())))
+                    .add("type", Json.createArrayBuilder()
+                            .add(jsonObjectBuilder().add("href", "https://schema.org/AboutPage"))
+                    );
+        }
+
+        if (authors != null) {
+            mandatory.add("author", authors);
+        }
+        if (licenseString != null && !Objects.equals(licenseString, "")) {
+            mandatory.add("license", jsonObjectBuilder().add("href", licenseString));
+        }
+        if (!mediaTypes.toString().trim().isEmpty()) {
+            mandatory.add("describedby", mediaTypes);
+        }
+        if (items != null) {
+            mandatory.add("item", items);
+        }
+        linksetJsonObj.add(mandatory);
+
+        // remove scholarly type as shown already on landing page
+        for (FileMetadata fm : fms) {
+            DataFile df = fm.getDataFile();
+            JsonObjectBuilder itemAnchor = jsonObjectBuilder().add("anchor", getPublicDownloadUrl(df));
+            itemAnchor.add("collection", Json.createArrayBuilder().add(jsonObjectBuilder()
+                    .add("href", landingPage)));
+            linksetJsonObj.add(itemAnchor);
+        }
+
+        return linksetJsonObj;
+    }
+
+
+    private String getPublicDownloadUrl(DataFile dataFile) {
+        StorageIO<DataFile> storageIO = null;
+        try {
+            storageIO = dataFile.getStorageIO();
+        } catch (IOException e) {
+            logger.warning(String.format("Error getting storageID from file; original error message is: %s", e.getLocalizedMessage()));
+        }
+
+        if (storageIO instanceof SwiftAccessIO) {
```

Comment:
I have a few questions about this:

- Given that files may be restricted, that an admin could move where files are stored to a different bucket or store type, etc., does it make sense to provide direct URLs to the storage? 
- Is Swift the only store that needs special processing? (S3, remote stores that are coming ~soon?)
- If needed, should this be optional? (Personally, I think I'd just send the Dataverse filedownload URL and let it handle access control and potential temporary/permanent redirects when that URL is accessed.)